python/ray/tests/test_node_manager.py

```diff
@@ -46,9 +53,57 @@ def f():
     ray.get([f._remote(args=[], kwargs={}, resources={str(i): 1}) for i in range(3)])
 
 
+@pytest.mark.parametrize(
+    "call_ray_start",
+    ["""ray start --head"""],
+    indirect=True,
+)
+def test_kill_driver_clears_backlog(call_ray_start):
+    driver = """
+import ray
+
+@ray.remote
+def f():
+    import time
+    time.sleep(300)
+
+refs = [f.remote() for _ in range(10000)]
+
+ray.get(refs)
+  """
+    proc = run_string_as_driver_nonblocking(driver)
+    ctx = ray.init(address=call_ray_start)
+
+    def check_has_backlog() -> bool:
+        resources_batch = get_resource_usage(
+            gcs_address=ctx.address_info["gcs_address"]
+        )
+        backlog = (
+            resources_batch.resource_load_by_shape.resource_demands[0].backlog_size
+            if resources_batch.resource_load_by_shape.resource_demands
+            else 0
+        )
+        return backlog > 0
+
+    wait_for_condition(check_has_backlog, timeout=10, retry_interval_ms=1000)
+
+    os.kill(proc.pid, 9)
+
+    def check_no_backlog() -> bool:
+        resources_batch = get_resource_usage(
```

Comment:
minor, but consider using the output of ray status instead? You can find how to query from https://github.com/ray-project/ray/blob/f8936f9c233abdf5c77894fa27410410c18d1947/dashboard/modules/reporter/reporter_head.py#L71

since demand seems to be more impl detail of ray status, and we'd like to make sure ray status has the correct output

Reply:
Is this a string output? I think its also fine to test the output of ray status just that the parsing code is more hairy and easy to break if we change it

Reply:
Btw I tried your api

```
// use rest client
webui_url = ctx.address_info["webui_url"]
webui_url = format_web_url(webui_url)
response = requests.get(f"{webui_url}/api/cluster_status")
response.raise_for_status()
print(response.json())

// use gcs client
address = ctx.address_info["gcs_address"]
gcs_client = ray._private.gcs_utils.GcsClient(address=address)
ray.experimental.internal_kv._initialize_internal_kv(gcs_client)
status = ray.experimental.internal_kv._internal_kv_get(
    ray_constants.DEBUG_AUTOSCALING_STATUS
)
```

but then this doesn't print out the demands:

`{'result': True, 'msg': 'Got cluster status.', 'data': {'autoscalingStatus': None, 'autoscalingError': None, 'clusterStatus': {'gcsRequestTime': 0.002483367919921875, 'loadMetricsReport': {'usage': {}, 'resourceDemand': [], 'pgDemand': [], 'requestDemand': [], 'nodeTypes': [], 'headIp': None, 'usageByNode': {}}, 'time': 1671231520.027012, 'monitorPid': 34915}}}
`

I did the same in test_dashboard and saw the same thing

Reply:
hmm that's odd..? this should be the same data as the result of `ray status`. Can you check if the result is there for normal cases (like when there are pending tasks)?

Reply:
I believe this field `resourceDemand ` should be available.